Mitmproxy android

RuneTM opened the pull request:
Added functionality to push the public key element of the keypair that mitmproxy uses to Android devices during initialization of emulator and agent. This way it should be possible to use mitmproxy to dump otherwise encrypted SSL/TLS traffic.

I know it is a little redundant to specify the public key in the auxiliary.conf configuration file, however it is significantly easier than extracting just the public key from the PKCS#12 keypair on the fly. Furthermore in due time, I would like to add functionality to Cuckoo where the CA certificate for mitmproxy is generated per analysis using parameters defined in a configuration file.

Finally to help decoding of SSL/TLS in Wireshark etc, this patch contains a small fix that make mitmproxy dump TLS master secrets to the analysis directory.

Commit:
Implement mitmproxy functionality for Android.
Additionally, make mitmdump dump its TLS master secrets to the analysis storage folder for cipher text decryption in Wireshark.

jbremer commented:
Sweet! This was on my todo list. I'll test it in a bit - thanks!

swackhamer commented on modules/auxiliary/mitm.py:
```diff
@@ -57,22 +66,31 @@ def start(self):
+        tls_secrets_path = os.path.join(CUCKOO_ROOT, "storage", "analyses",
+                                 "%d" % self.task.id, "mitm_tls_secrets.log")
+
+        self.proc = subprocess.Popen(shlex.split(command),
```
arguments passed into Popen should probably be a list and not a split string in cases like this where you can control the arguments.

Commit:
Undo shlex splitting of mitmdump command

Commit:
Revert "Fix for race condition where agent is requested before it is ready"

This reverts commit a7f9551211dbcbb9e8f58ba559c653bd4d818ef2.

RuneTM commented:
I reverted the changes to how the mitmdump command is structured for popen, as it actually works without it, and it was probably only needed because when I started implementing this functionality I was using an old version of mitmproxy.

Also I reverted the other "time.sleep(10)" commit  from this pull request, as it is a separate pull request, sorry about that.

jbremer commented:
You read my mind - thanks! Will check this out later.

kuza55 commented:
It looks like this code got added and then reverted; does anyone know what the issue was that caused it to get pulled?

jbremer commented:
Did anyone have the time to test this PR out by any chance? It looks quite fine, overall, but haven't had time myself.
